curriculum/challenges/english/25-front-end-development/lab-checkout-page/66da326c02141df538f29ba5.md

```diff
@@ -0,0 +1,175 @@
+
+```js
+assert.equal(document.querySelector('h1')?.innerText, 'Checkout');
+```
+
+You should only have one `h1` element on your page.
+
+```js
+assert.lengthOf(document.querySelectorAll('h1'), 1);
+```
+
+You should have at least two `section` elements after the `h1` element.
+
+```js
+const h1 = document.querySelector('h1');
+const firstSibling = h1?.nextElementSibling;
+const secondSibling = firstSibling?.nextElementSibling;
+assert.equal(firstSibling?.tagName, 'SECTION');
+assert.equal(secondSibling?.tagName, 'SECTION');
+```
+
+You should have an `h2` with the text `Your Cart` within the first section.
+
+```js
+const firstSection = document.querySelectorAll('section')?.[0];
+assert.equal(firstSection?.querySelector('h2')?.innerText, 'Your Cart');
+```
+
+You should have at least one image with alternate text in your first section.
+
+```js
+const firstSection = document.querySelectorAll('section')?.[0];
+assert.isAtLeast(firstSection?.querySelector('img')?.alt?.length, 1);
+```
+
+You should have an `h2` with the text `Payment Information` within the second section.
+
+```js
+const secondSection = document.querySelectorAll('section')?.[1];
+assert.equal(secondSection?.querySelector('h2')?.innerText, 'Payment Information');
+```
+
+You should have a `form` within the second section.
+
+```js
+const secondSection = document.querySelectorAll('section')?.[1];
+assert.exists(secondSection?.querySelector('form'));
+```
+
+You should have an input with an `id` and `name` of `card-name` within your form.
+
+```js
+const secondSection = document.querySelectorAll('section')?.[1];
+assert.exists(secondSection?.querySelector('form input#card-name'));
+assert.exists(secondSection?.querySelector('form input[name="card-name"]'));
+```
+
+You should have an input with an `id` and `name` of `card-number` within your form.
+
+```js
+const secondSection = document.querySelectorAll('section')?.[1];
+assert.exists(secondSection?.querySelector('form input#card-number'));
+assert.exists(secondSection?.querySelector('form input[name="card-number"]'));
+```
+
+All of your input elements that aren't a `type` of `submit` should have a label associated with them.
+
+```js
+const inputs = document.querySelectorAll('input');
+inputs.forEach(input => {
+  if (input.type !== 'submit') {
+    assert.exists(document.querySelector(`label[for="${input.id}"]`))
+  }
+});
+```
+
+At least two of your input elements should be required and have an `aria-required` attribute set to `true`.
+
+```js
+const inputs = document.querySelectorAll('input');
+const requiredInputs = Array.from(inputs).filter(input => input.required && input.getAttribute('aria-required') === 'true');
+assert.isAtLeast(requiredInputs?.length, 2)
+```
+
+You should use `tabindex` at least once.
+
+```js
+const els = document.querySelectorAll('[tabindex]');
+console.log(els);
+assert.isAtLeast(els.length, 1);
+```
+
+# --seed--
+
+## --seed-contents--
+
+```html
+
```

Comment:
```suggestion
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Checkout Page</title>
</head>
<body>

</body>
</html>
```

Reply:
The lab before this has already the boiler in the seed code.

Reply:
✅